miner/worker.go

```diff
@@ -424,23 +417,29 @@ func (self *worker) commitNewWork() {
 		tstamp = parent.Time().Int64() + 1
 	}
 	// this will ensure we're not going off too far in the future
-	if now := time.Now().Unix(); tstamp > now+4 {
+	if now := time.Now().Unix(); tstamp > now+1 {
```

Comment:
There's also a drawback, that if the miner "sleeps" for 4 seconds (which happened previously), but blocks keep pouring in, a lot of weird things could happen.

We should really rewrite the miner.